tokio/src/net/tcp/stream.rs

```diff
@@ -1015,6 +1015,48 @@ impl TcpStream {
             .registration()
             .try_io(interest, || self.io.try_io(f))
     }
+    
+    /// Reads or writes from the socket using a user-provided IO operation.
+    ///
+    /// The readiness of the socket is awaited and when the socket is ready,
+    /// the provided closure is called. The closure should attempt to perform
+    /// IO operation on the socket by manually calling the appropriate syscall.
+    /// If the operation fails because the socket is not actually ready,
+    /// then the closure should return a `WouldBlock` error, the readiness
+    /// flag is cleared and the socket readiness is awaited again. This loop
+    /// repeated until the closure returns an `Ok` or an error that doesn't
+    /// have the `WouldBlock` value.
+    ///
+    /// The closure should only return a `WouldBlock` error if it has performed
+    /// an IO operation on the socket that failed due to the socket not being
+    /// ready. Returning a `WouldBlock` error in any other situation will
+    /// incorrectly clear the readiness flag, which can cause the socket to
+    /// behave incorrectly.
+    ///
+    /// The closure should not perform the IO operation using any of the methods
+    /// defined on the Tokio `UdpSocket` type, as this will mess with the
```

Comment:
Here it should refer to the `TcpStream`, as this is the type we are adding a method for. The same goes for other types.
```suggestion
    /// defined on the Tokio `TcpStream` type, as this will mess with the
```